### PR TITLE
Add configurable timeouts to http server

### PR DESCRIPTION
**Pull request by yihuang**
## Description

Closes: #963

not tested yet.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: server/config/toml.go [hunk 1]
```diff
@@ -53,6 +53,12 @@ logs-cap = {{ .JSONRPC.LogsCap }}
 # BlockRangeCap defines the max block range allowed for 'eth_getLogs' query.
 block-range-cap = {{ .JSONRPC.BlockRangeCap }}
 
+# HTTPTimeout is the read/write timeout of http json-rpc server.
+http-timeout = "{{ .JSONRPC.HTTPTimeout }}"
+
+# HTTPIdleTimeout is the idle timeout of http json-rpc server.
+http-idle-timeout = "{{ .JSONRPC.HTTPIdleTimeout }}"
```
Comment (fedekunze) [inline server/config/toml.go:60]: the value should not be a string

```suggestion
http-timeout = {{ .JSONRPC.HTTPTimeout }}

# HTTPIdleTimeout is the idle timeout of http json-rpc server.
http-idle-timeout = {{ .JSONRPC.HTTPIdleTimeout }}
```

**Review by fedekunze — COMMENTED** [review]
Minor comment, I'd also add a test so that we ensure that the toml is correctly parsed to the config

File: server/config/toml.go (hunk 1 above)
Comment (yihuang) [reply server/config/toml.go:60]: we need to quote, because duration is serialized sth like `"10s"`.

Comment (fedekunze) [reply server/config/toml.go:60]: ah right

**Review by fedekunze — APPROVED** [review]

**Pull request merged**